dask/array/gufunc.py

```diff
@@ -142,13 +145,11 @@ def apply_gufunc(func, signature, *args, **kwargs):
     ## Determine nout: nout = None for functions of one direct return; nout = int for return tuples
     nout = None if not isinstance(core_output_dimss, list) else len(core_output_dimss)
 
-    ## Assert output_dtypes
+    ## Determine and handle output_dtypes
     if output_dtypes is None:
-        raise ValueError("Must specify `output_dtypes` of output array(s)")
-    elif isinstance(output_dtypes, str):
-        otypes = list(output_dtypes)
-        output_dtypes = otypes[0] if nout is None else otypes
-    elif isinstance(output_dtypes, (tuple, list)):
+        output_dtypes = apply_infer_dtype(func, args, kwargs, "apply_gufunc", "output_dtypes", nout)
+
+    if isinstance(output_dtypes, (tuple, list)):
         if nout is None:
             raise ValueError("Must specify single dtype for `output_dtypes` for function with one output")
```

Comment:
Can you verify that supplying a list of one dtype works?

Reply:
This behavior has been here since original PR #3109. Currently an error is raised, because for a single output the dtype can be given directly as dtype without list. E.g.
```python
def foo():
    return 1

apply_gufunc(foo, "->()", output_dtypes=int)
```

It is related to our [previous discussion](https://github.com/dask/dask/pull/3109#discussion_r164537146), where I have raised numpy/numpy#10527 as general issue. The outcome of our discussion has been to fully stick to numpy signature (no trailing commas in output, i.e. no tuple of length as return of functions).

Per your request, I am going to enable this behavior. IMHO, as for aforementioned reasons,  I still think it is inconsistent to what Python provides as possibilities to return values.

Reply:
I agree that NumPy API for gufuncs has some unfortunate inconsistencies, but I think it's a little different from the `output_dtypes` argument here -- which is not something that's directly in NumPy's API.